packages/roosterjs-content-model/lib/modelApi/common/mergeModel.ts

```diff
@@ -259,30 +263,56 @@ function insertBlock(markerPosition: InsertPoint, block: ContentModelBlock) {
     }
 }
 
-function applyDefaultFormat(group: ContentModelBlockGroup, format: ContentModelSegmentFormat) {
+function applyDefaultFormat(
+    group: ContentModelBlockGroup,
+    format: ContentModelSegmentFormat,
+    applyDefaultFormatOption: 'MergeCurrentFormat' | 'ApplyCurrentFormat'
+) {
     group.blocks.forEach(block => {
         switch (block.blockType) {
             case 'BlockGroup':
-                applyDefaultFormat(block, format);
+                applyDefaultFormat(block, format, applyDefaultFormatOption);
                 break;
 
             case 'Table':
                 block.rows.forEach(row =>
                     row.cells.forEach(cell => {
-                        applyDefaultFormat(cell, format);
+                        applyDefaultFormat(cell, format, applyDefaultFormatOption);
                     })
                 );
                 break;
 
             case 'Paragraph':
                 block.segments.forEach(segment => {
                     if (segment.segmentType == 'General') {
-                        applyDefaultFormat(segment, format);
+                        applyDefaultFormat(segment, format, applyDefaultFormatOption);
                     }
 
-                    segment.format = { ...format, ...segment.format };
+                    segment.format =
+                        applyDefaultFormatOption == 'MergeCurrentFormat'
+                            ? { ...format, ...segment.format }
+                            : {
+                                  ...format,
+                                  ...getSemanticFormat(segment),
+                              };
                 });
                 break;
         }
     });
 }
+
+function getSemanticFormat(segment: ContentModelSegment): ContentModelSegmentFormat {
+    const result: ContentModelSegmentFormat = {};
+
+    if (segment.format.fontWeight) {
```

Comment:
What if fontWeight is "normal", what is the expected result in that case? Especially the current format has fontWeight to be bold?

Reply:
Good point, I checked Word and if the fontWeight is normal or initial, we should not use the font weight from the source.
Will add this check here.